packages/schematics/src/app/utils.ts

```diff
@@ -0,0 +1,207 @@
+import { JsonAstObject, JsonParseMode, normalize, parseJsonAst } from '@angular-devkit/core';
+import {
+  apply,
+  branchAndMerge,
+  mergeWith,
+  move,
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  template,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { getWorkspace, updateWorkspace } from '@schematics/angular/utility/config';
+import {
+  addPackageJsonDependency,
+  NodeDependency,
+  NodeDependencyType,
+} from '@schematics/angular/utility/dependencies';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+  insertPropertyInAstObjectInOrder,
+} from '@schematics/angular/utility/json-utils';
+import { getProject } from '@schematics/angular/utility/project';
+import { getProjectTargets } from '@schematics/angular/utility/project-targets';
+import { of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+import { getLatestNodeVersion, NpmRegistryPackage } from './npmjs-utils';
+import { Schema } from './schema';
+
+export function validateOptions(options: Schema): Rule {
```

Comment:
Wondering if it would make sense to organize these into a `rules` directory rather then in a big `utils` file?

Reply:
Maybe instead of a directory I could rename the files to be `rules.ts`?